store/tikv/region_cache.go

```diff
@@ -291,13 +323,18 @@ func (c *RegionCache) getCachedRegion(id RegionVerID) *Region {
 // searchCachedRegion finds a region from cache by key. Like `getCachedRegion`,
 // it should be called with c.mu.RLock(), and the returned Region should not be
 // used after c.mu is RUnlock().
-func (c *RegionCache) searchCachedRegion(key []byte) *Region {
+// If the given key is the end key of the region that you want, you may set the second argument to true. This is useful when processing in reverse order.
+func (c *RegionCache) searchCachedRegion(key []byte, isEndKey bool) *Region {
 	var r *Region
 	c.mu.sorted.DescendLessOrEqual(newBtreeSearchItem(key), func(item btree.Item) bool {
 		r = item.(*btreeItem).region
+		if isEndKey && bytes.Compare(r.StartKey(), key) == 0 {
```

Comment:
em, how about use `DescendLessThan` instead of `DecendLessOrEqual` when reverse seek?